### PR TITLE
feat: `cmake` port.

### PR DESCRIPTION
**Pull request by destifo**
### Describe your change

- Add a `cmake` Port

### Motivation and context

[MET-487](https://linear.app/metatypedev/issue/MET-487/cmakeasdf-is-broken-on-mac)

### Migration notes

_No Migrations Needed_

### Checklist

- [x] The change come with new or modified tests
- [x] Hard-to-understand functions have explanatory comments
- [x] End-user documentation is updated to reflect the change

### CODE REVIEW

File: examples/cmake/ghjk.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const allowedPortDeps = [
+  ...stdDeps(),
+  ...[installs.python_latest].map((fat) => ({
+    manifest: fat.port,
+    defaultInst: thinInstallConfig(fat),
+  })),
+];
+
+export const secureConfig = ssecureConfig({ allowedPortDeps });
```
Comment (zifeo) [inline examples/cmake/ghjk.ts:22]: does not seem to work without secureConfig, any idea why?
```
Uncaught (in promise) TypeError: Cannot destructure property 'manifest' of 'portsConfig.allowedDeps[depId.name]' as it is undefined.
```

File: examples/cmake/ghjk.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const PYTHON_VERSION = "3.8.18";
+
+const installs = {
+  python: ports.cpy_bs({ version: PYTHON_VERSION, releaseTag: "20240224" }),
+  python_latest: ports.cpy_bs({ releaseTag: "20240224" }),
+};
+
+const allowedPortDeps = [
+  ...stdDeps(),
+  ...[installs.python_latest].map((fat) => ({
```
Comment (zifeo) [inline examples/cmake/ghjk.ts:16]: [x].map?

File: examples/cmake/ghjk.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  })),
+];
+
+export const secureConfig = ssecureConfig({ allowedPortDeps });
+// export const secureConfig = ghjk.secureConfig({
+//     allowedPortDeps: [...ghjk.stdDeps({ enableRuntimes: true })],
+//   });
+
+install(
+  ports.pipi({
```
Comment (zifeo) [inline examples/cmake/ghjk.ts:28]: works, let's define this as an alias for cmake directly in the cmake port

**Review by zifeo — CHANGES_REQUESTED** [review]

File: examples/cmake/ghjk.ts (hunk 2 above)
Comment (Yohe-Am) [reply examples/cmake/ghjk.ts]: No longer relevant after #87.

File: examples/cmake/ghjk.ts (hunk 1 above)
Comment (Yohe-Am) [reply examples/cmake/ghjk.ts]: This was a bad error message case when a required dep port wasn't added to the allow set. Solved in #91.

**Comment by Yohe-Am** [comment]
I've added the fixes for ruff 0.5.0 in #94 PR which should merge soon.

File: ports/cmake.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { InstallConfigFat, InstallConfigSimple } from "../port.ts";
+import { AsdfInstallConf } from "./asdf.ts";
+import { PipiInstallConf } from "./pipi.ts";
+import * as ports from "./mod.ts";
+
+export default function conf(
+  config: InstallConfigSimple,
+): InstallConfigFat[] {
+  if (Deno.build.os === "darwin") {
```
Comment (zifeo) [inline ports/cmake.ts:9]: Let's add a comment on why we have this, otherwise your future you will forget :)

**Review by zifeo — APPROVED** [review]

**Pull request merged**